File: Programming/C/CS50x/week6/sentimental-readability/readability.py
from cs50 import get_string
import re


def compute_index(text):
    letters_num = len(re.findall(r"[A-Za-z]", text))
    words_num = len(text.split())
    sentences_num = len(re.findall(r'[\.!?]', text))
    print(f"letters_num: {letters_num} words_num: {words_num} sentences_num: {sentences_num}")
    index = 0.0588 * letters_num / words_num * 100 - 0.296 * sentences_num / words_num * 100 - 15.8
    return index


text = get_string("Text: ")
index = compute_index(text)
grade = round(index)
print(index)
if index < 1.5:
    print("Before Grade 1")
elif index >= 15.5:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: Courses/CS50x/week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

float compute_index(string sentences);

int main(void)
{
    string sentences = get_string("Text: ");
    float index = compute_index(sentences);
    if (index < 1.5)
        printf("Before Grade 1\n");
    else if (index >= 15.5)
        printf("Grade 16+\n");
    else
        printf("Grade %0.f\n", index);
    return 0;
}

float compute_index(string sentences)
{
    float index = 0;
    int words_num = 1;
    int sentences_num = 0;
    int letters_num = 0;
    for (int i = 0; sentences[i] != '\0'; i++)
    {
        if (isalpha(sentences[i]))
            letters_num++;
        else if (sentences[i] == ' ')
            words_num++;
        else if ((sentences[i] == '.' || sentences[i] == '?') || sentences[i] == '!')
            sentences_num++;
    }
    index = 0.0588 * letters_num / words_num * 100 - 0.296 * sentences_num / words_num * 100 - 15.8;
    return index;
}
